src/java/org/apache/cassandra/cql3/CQLStatement.java

```diff
@@ -32,9 +34,9 @@ public interface CQLStatement
     /**
      * Returns all bind variables for the statement
      */
-    default List<ColumnSpecification> getBindVariables()
+    default ImmutableList<ColumnSpecification> getBindVariables()
     {
-        return Collections.emptyList();
```

Comment:
Why switching this to immutable list? emptyList() would have same allocation profile right?

Reply:
The problem is with ImmutableList.copyOf(columnSpecs) within OptionsWithColumnSpecifications constructor, we have to copy columnSpecs for every request because in one place plain List is used and in another - ImmutableList